src/main/java/org/xwiki/git/script/GitScriptService.java

```diff
@@ -99,6 +102,44 @@ public Repository getRepository(String repositoryURI, String localDirectoryName,
         return this.gitManager.getRepository(repositoryURI, localDirectoryName, username, accessCode);
     }
 
+    /**
+     * Clone a Git repository as Bare by storing it locally in the XWiki Permanent directory. If the repository is
+     * already cloned, no action is done.
+     *
+     * @param repositoryURI the URI to the Git repository to clone (eg "git://github.com/xwiki/xwiki-commons.git")
+     * @param localDirectoryName the name of the directory where the Git repository will be cloned (this directory is
+     *        relative to the permanent directory
+     * @return the cloned Repository instance
+     * @since 9.10
+     */
+    @Unstable
+    public Repository getRepositoryBare(String repositoryURI, String localDirectoryName)
+    {
+        CloneCommand cloneCommand = Git.cloneRepository();
+        return this.gitManager.getRepositoryBare(repositoryURI, localDirectoryName, cloneCommand);
+    }
+
+    /**
+     * Clone a private Git repository as Bare using the credentials provided by user and store it locally in the
+     * XWiki Permanent directory. If the repository is already cloned, no action is done.
+     *
+     * @param repositoryURI the URI to the Git repository to clone (eg "git://github.com/xwiki/xwiki-commons.git")
+     * @param localDirectoryName the name of the directory where the Git repository will be cloned (this directory is
+     *        relative to the permanent directory
+     * @param username the username of the Git user
+     * @param accessCode the password or OAuth or personal access token that authenticates with the Git user
+     * @return the cloned Repository instance
+     * @since 9.10
+     */
+    @Unstable
+    public Repository getRepositoryBare(String repositoryURI, String localDirectoryName, String username,
+        String accessCode)
+    {
+        CloneCommand cloneCommand = Git.cloneRepository();
+        cloneCommand.setCredentialsProvider(new UsernamePasswordCredentialsProvider(username, accessCode));
+        return this.gitManager.getRepositoryBare(repositoryURI, localDirectoryName, cloneCommand);
+    }
+
```

Comment:
You're missing the generic `getRepository` signature accepting a `CloneCommand`. Right now you cannot call it from a script which is not good. Note that this since it's not possible to create objects from Velocity, we probably need a `createCloneCommand()` method too.

Reply:
I don't think we should add the `CloneCommand` here *atm*. It would be good yes, but referring to this specific issue (bare cloning), we should add `getRepositoryBare` and later on we can add `getRepository` with the `CloneCommand`. I mean, if someone wants only the bare use case, he won't have to go through `CloneCommand`.
What do you say?

Reply:
I say that bare cloning is not the only use case, so I would still add a `createCloneCommand()`  + the getRepo siganture acccepting a CloneCommand.

Reply:
Ok, I have done all the changes except this one which is what I am going to do today.
How should I make the `createCloneCommand()`. Like `createCloneCommand(setBare, creds...)` or can  you suggest a better way? 
Thanks.

Reply:
`createCloneCommand()`.

Reply:
Got it.

Reply:
@vmassol  I have commited changes.
For this specifically: I have created `createCloneCommand()` in GitScriptService. I can create it in the `GitManager` interface and implement it in `DefaultGitManager` and use that one instead, but I am unsure if I should.